src/pg/sql/clauses.js

```diff
@@ -66,7 +89,13 @@ export const getSelectCols = (options, projection = null) => {
         sql`jsonb_build_object(${join(subSqls, ', ')}) AS "${raw(key)}"`,
       );
     } else {
-      sqls.push(sql`"${raw(key)}"`);
+      if (typeof projection[key] === 'object') {
+        keyPathGenerator({ [key]: projection[key] }).forEach((path) => {
+          sqls.push({"values":[],"strings":[path]});
+        })
+      } else {
+        sqls.push(sql`"${raw(key)}"`);
+      }
```

Comment:
This is on the right path, but it can be simplified. My suggestion:

```suggestion
      // The recursive function will handle checking whether the projection
      // is an object, so we don't need to do that here.
      const colSql = getColSql([key], projection[key], options);
      
      // As it is not always a simple column lookup, we need to use SQL `AS`
      // to name the returned column correctly; Graffy won't know what to do
      // with a property named "jsonb_build_object"
      sqls.push(sql`${colSql} AS "${raw(key)}"`);
```

The getColSql function can be:

```js
function getColSql(path, projection, options) {
  if (!object || typeof projection !== 'object') return lookup(path, options);

  // To pass to jsonb_build_object, we need to flatten the object into
  // an array [key1, value1, key2, value2, ...]
  const kv = Object.entries(projection).flatMap(([key, val]) => [
    key,
    getColSql(path.concat(key), projection[key], options)
  ]);

  return sql`jsonb_build_object(${join(kv, ', ')})`;
}
```

Note that we use the existing lookup() function to avoid writing the `->` expressions manually.